runners/core-java/src/main/java/org/apache/beam/runners/core/SideInputHandler.java

```diff
@@ -163,11 +164,6 @@ public void addSideInputValue(
   @Override
   public <T> T get(PCollectionView<T> sideInput, BoundedWindow window) {
 
-    if (!isReady(sideInput, window)) {
```

Comment:
The real check about whether to process main-input elements or not is done here: https://github.com/apache/beam/blob/master/runners/core-java/src/main/java/org/apache/beam/runners/core/PushbackSideInputDoFnRunner.java#L83

In reality, if this check here fails it's already too late. Tests were failing because we didn't process main-input data for missing side inputs. This change makes sure that we process pushed back elements when we get the `+Inf` watermark on the side input: https://github.com/apache/beam/pull/2263/files#diff-bb49b4e5c16cd7d4f80a489170f7851aR482. For this to work we needed to remove this check to allow getting an empty side input.